gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/right_click_detection/process.mcfunction

```diff
@@ -0,0 +1,23 @@
+# processes the "right click detection" interaction so that it is despawned when unneeded
+# @s = gm4_llp_placement_rcd interaction 
+# at @s
+# run from mechanics/right_click_detection/loop
+
+# visuals
+particle scrape ~ ~.1 ~ 0 0 0 0 1
+
+# refresh timer
+scoreboard players set $timer gm4_llp.data 10
+
+# kill rcd if uneeded
+execute unless entity @a[tag=gm4_llp_holding_item,distance=..6] run return run kill @s
+execute unless block ~ ~ ~ lily_pad run return run kill @s
+execute if entity @e[type=block_display,tag=gm4_llp_display,distance=..0.1] run return run kill @s
+
+# kill if multiple interaction with the same id, which therefore belong to the same player
+tag @s add gm4_llp.rcd_to_check
+scoreboard players set $duplicate_exists gm4_llp.data 0
+execute as @e[type=interaction,tag=gm4_llp_placement_rcd,tag=!gm4_llp.rcd_to_check,distance=0.1..16] \
+  if score @s gm4_llp.id = @e[type=interaction,tag=gm4_llp.rcd_to_check,distance=..0.1,limit=1] gm4_llp.id \
+  run scoreboard players set $duplicate_exists gm4_llp.data 1
```

Comment:
I'd prefer it if you stored the id score of this interaction into a fake player and then compared the other interaction's score against that fake player to avoid a lot of entity lookups.